### PR TITLE
thompson first try

### PR DESCRIPTION
**Pull request by ssikdar1**

### CODE REVIEW

File: simulation_code/simulations.jl [hunk 1]
```diff
@@ -228,7 +232,10 @@ end
         mu_Uit = mu_Vit + beta * cur_V
         # make choice
         ce_Uit = certainty_equivalent(alpha, mu_Uit, Sigma_Vit) # γ: uncertainty aversion
+
         c_it = nothing
+        if use_thompson
+            c_it = argmax(rand(V))
         if rand() < epsilon
```
Comment (rawls238) [inline simulation_code/simulations.jl]: can we just kill the epsilon greedy thing?

File: simulation_code/simulations.jl [hunk 2]
```diff
@@ -274,8 +287,11 @@ end
         # make choice
         ce_Uit = certainty_equivalent(alpha, mu_Uit, Sigma_Uit)
         choice_set = [n for n in Nset if n ∉ C_iT]
+        
         c_it = 0
-        if rand() < epsilon
+        if use_thompson
+            c_it = argmax(rand(U_i)) 
+        elseif rand() < epsilon
```
Comment (rawls238) [inline simulation_code/simulations.jl]: same as above

File: simulation_code/simulations.jl [hunk 3]
```diff
@@ -341,7 +357,7 @@ end
     mu_V = zeros(Float64, N)
     V = rand(MvNormal(mu_V, Sigma_V))
 
-
+    use_thompson::Bool = true
```
Comment (rawls238) [inline simulation_code/simulations.jl]: this should definitely be a function argument and not hardcoded in the function

File: simulation_code/test.jl [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+function thompson_sampling(
+    mu_V::Array{Float64,1},
+    Sigma_V::Array{Float64,2},
+    N
+)
+
+
+    draws = [
+            rand(
+                MvNormal(
```
Comment (rawls238) [inline simulation_code/test.jl]: this shouldn't be multivariate Normal but rather just a univariate Normal

File: simulation_code/test.jl [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                    [mu_V[ii]],
+                    [Sigma_V[ii,ii]]
+                )
+            )  for ii in 1:N ]
+    c_it_index = argmax(draws)
+    c_it = draws[c_it_index][1]
```
Comment (rawls238) [inline simulation_code/test.jl]: we should return the index, not the draw. The draw isn't important

File: simulation_code/simulations.jl [hunk 6]
```diff
@@ -199,6 +199,25 @@ end
...
+    
+    draws = [ 
+            rand(
+                MvNormal(
+                    [mu_V[ii]], 
+                    [Sigma_V[ii,ii]]
+                )
+            )  for ii in 1:N ]
+    c_it_index = argmax(draws)
+    c_it = draws[c_it_index][1]
```
Comment (rawls238) [inline simulation_code/simulations.jl]: ditto as below

File: simulation_code/simulations.jl [hunk 7]
```diff
@@ -228,8 +248,11 @@ end
         mu_Uit = mu_Vit + beta * cur_V
         # make choice
         ce_Uit = certainty_equivalent(alpha, mu_Uit, Sigma_Vit) # γ: uncertainty aversion
+
         c_it = nothing
-        if rand() < epsilon
+        if use_thompson
+            c_it = thompson_sampling(mu_V_i, Sigma_V_i, N)
```
Comment (rawls238) [inline simulation_code/simulations.jl]: so what we should get returned here is the index of the item to be cnosumed, not the draw

File: simulation_code/simulations.jl [hunk 8]
```diff
@@ -274,8 +299,12 @@ end
         # make choice
         ce_Uit = certainty_equivalent(alpha, mu_Uit, Sigma_Uit)
         choice_set = [n for n in Nset if n ∉ C_iT]
+        
         c_it = 0
-        if rand() < epsilon
+        if use_thompson
+            c_it = thompson_sampling(mu_U_i, Sigma_U_i, N)
```
Comment (rawls238) [inline simulation_code/simulations.jl]: no need to pass in `N`

File: simulation_code/simulations.jl [hunk 9]
```diff
@@ -199,6 +199,25 @@ end
...
+)
+
+    
+    draws = [ 
+            rand(
+                MvNormal(
+                    [mu_V[ii]], 
+                    [Sigma_V[ii,ii]]
+                )
+            )  for ii in 1:N ]
```
Comment (rawls238) [inline simulation_code/simulations.jl]: you can just have in 1:length(mu_V)

Comment (ssikdar1) [reply simulation_code/simulations.jl]: fixed in  b089f70

File: simulation_code/simulations.jl (hunk 7 above)
Comment (ssikdar1) [reply simulation_code/simulations.jl]: changed `c_it` to be 
```
c_it = argmax(draws)
```
in b089f70

File: simulation_code/test.jl (hunk 5 above)
Comment (ssikdar1) [reply simulation_code/test.jl]: addressed in b089f70

File: simulation_code/test.jl (hunk 4 above)
Comment (ssikdar1) [reply simulation_code/test.jl]: changed to `Normal` in b089f70

File: simulation_code/simulations.jl [hunk 10]
```diff
@@ -274,8 +297,12 @@ end
         # make choice
         ce_Uit = certainty_equivalent(alpha, mu_Uit, Sigma_Uit)
         choice_set = [n for n in Nset if n ∉ C_iT]
+        
         c_it = 0
-        if rand() < epsilon
+        if use_thompson
+            c_it = thompson_sampling(mu_U_i, Sigma_U_i)
+            @show c_it
```
Comment (rawls238) [inline simulation_code/simulations.jl]: this is probably useful for debugging but let's kill this before merge?

**Comment by rawls238** [comment]
lgtm

File: simulation_code/simulations.jl (hunk 10 above)
Comment (ssikdar1) [reply simulation_code/simulations.jl]: resolved in 80e579c

**Pull request merged**